File: canvas/fizz-11/lib/assignments.js

const axios = require('./axios')

const getAssignments = async (courseId, assignmentStatus) => {
  const { data } = await axios.get(`/api/v1/courses/${courseId}/assignments?bucket=${assignmentStatus}`)
  return data
}

const getAssignmentsForCourse = async (courseId) => {

  const pastAssignments = await getAssignments(courseId, 'past')
  const overdueAssignments = await getAssignments(courseId, 'overdue')

  return [ ...pastAssignments, ...overdueAssignments ]

}

const getMissingAssignmentsForUser =  async (userId) => {
  const { data } = await axios.get(`/api/v1/users/${userId}/missing_submissions`)
  return data
}



module.exports = {
  getAssignmentsForCourse,
  getMissingAssignmentsForUser
}

File: src/canvas/data-extractor/lib/assignments.py
from lib.canvasRequest import get

def getAssignments(courseId, assignmentStatus):
    assignments = get(f"/api/v1/courses/{courseId}/assignments?bucket={assignmentStatus}")
    return assignments

def getAssignmentsForCourse(courseId):
    past_assignments = getAssignments(courseId, "past")
    overdue_assignments = getAssignments(courseId, "overdue")
    return list(past_assignments + overdue_assignments)
